### ISSUE TITLE
Fewer topics than the number of articles

### ISSUE THREAD

**Issue by donottakemyusername** [MaartenGr/BERTopic#1427]
Hi, I am using the partial_fit function to generate topics continuously. However at some batches the number of topics could be a bit fewer than the number of articles in the dataset. I have checked the "labels" attribute in "partial_fit" and it returns exactly the same topics as articles, but when I called "model.topics_" after BERTopic it returns fewer topics. How could this happen and what can I do to solve it? Or at least, can I know which articles did not get assigned a topic? I can drop them directly as there are usually only 2 or 3 articles with missing topics. Thanks!

**Comment by MaartenGr** [comment]
It is difficult to say without seeing your code. Could you share a full example? Also, did you make sure to iteratively update `model.topics_` as is mentioned in the docs [here](https://maartengr.github.io/BERTopic/getting_started/online/online.html#example)?

**Comment by donottakemyusername** [comment]
I am using River to continuously generate new topics. Do I still need to update the model.topics_ parameter?

**Comment by MaartenGr** [comment]
Definitely! The `topics_` are only saved for each batch in each `.partial_fit` call. So if you want to save every topic for every document you have seen thus far, you will need to update it as follows:

![image](https://github.com/MaartenGr/BERTopic/assets/25746895/0b83e7c9-9c37-470b-9c1f-027d0339ab3a)

**Comment by donottakemyusername** [comment]
but aren't these topics already saved in the "topics" list? Currently inside each batch, I am calling:
df = pd.DataFrame({'Document': df['text'],
                                 'Topic': topics})
to create a dataframe and topics for that batch. Then i saved the model and data locally and cleared up the memory. Then I load the model again with a new batch of data and run the previous model. Does this work? I think for most of the batches it looks fine, it just generated 2 or 3 fewer topics than the number of articles seldomly.

**Comment by MaartenGr** [comment]
You should do it as I described with the picture above. If you did not, could you provide your full code? 

It might also be an issue with River but I am not entirely sure. Try incremental kmeans instead and see if that fixes your problem.

**Comment by donottakemyusername** [comment]
I see, I will try it. Thank you!
